text/0020-distingush_manual_vs_scripted_transforms.md

```diff
@@ -0,0 +1,312 @@
+
+`dataset.yaml`:
+```yaml
+structure:
+  format: json
+  schema:
+    type: array
+    items:
+      type: object
+      properties:
+        name:
+          type: string
+        description:
+          type: string
+          maxLength: 140
+```
+
+And run save with `--recall-tf`:
+
+```
+$ qri save --file=dataset.yaml --recall-tf me/ca_prime_ministers
+🤖 executing transform
+✅ transform complete
+updated dataset b5/ca_prime_minsters
+```
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Human and Machine Transforms that affect the same field is now an error
+
+While this RFC places the definition at the field level, our initial implemenation will operate at the component level for simplicity.
+
+This does mean we'll need tools to check which components & fields of a dataset a transform affects. Thanks to the way starlark works (and our implementation), this'll be pretty simple.
+
+Ways to resolve the error:
+* adjust your commit so that it doesn't affect
+* `--drop-transform` will remove the transform moving forward
+
+## The job of a transform script is to generate the next state snapshot from the existing snapshot
+This means the dataset passed into `transform(ds,ctx)` will be the existing dataset in Qri. Because human and machine transforms are mutually exclusive, script authors only need to consider which fields they wish to remain editable by users.
+
+
+## `save` is for humans, `update` is for machines
+Save is now the more powerful of the two commands. 
+
+By default save will _not_ recall and re-run transforms, but will run provided transforms. running `qri save` without providing a tra
+
+Local updates will need some refinement. Running `qri update me/dataset` will now only succeed if a 
+Update gets a new flag: `--recall-tf`
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main reason for not doing this is we haven't vetted this approach well enough.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The alternative is what we're currently doing, which is a sort of simultaneous three-way merge between the previous commit, user input, and script results. A more realistic alternative would be further refinements on this model, which we'll hopefully figure out through use.
+
+# Prior art
+[prior-art]: #prior-art
+
+[Git Hooks](https://githooks.com/) were mentioned as an example of prior art. There isn't much to go on here.
+There is no analog for a transform script in git. Git hooks are the closest concept, but only allow the user to receive notifications and interrupt the commit process, not affect the commit itself
+
```

Comment:
Also, macros in spreadsheet programs like Excel, Google Sheets, and OpenOffice have some analogue to what we're doing.